pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/HashJoinOperator.java

```diff
@@ -260,6 +260,7 @@ private void buildBroadcastHashTable()
         hashCollection.add(row);
       }
       _currentRowsInHashTable += container.size();
+      sampleAndCheckInterruption();
```

Comment:
IICU We are only calling `sampleAndChekInterruption` when the hash table is being build, but we should also call it whenever new blocks are received from the left hand side. Specifically, we may be creating very large maps when using cross join functions, but anyway the correct thing would be to call this method on every join type whenever a block is being processed

Reply:
Done. I am calling the function in `buildJoinDataBlock` as well.